pkg/kubelet/lifecycle/predicate.go

```diff
@@ -168,3 +172,68 @@ func removeMissingExtendedResources(pod *v1.Pod, nodeInfo *schedulernodeinfo.Nod
 	}
 	return podCopy
 }
+
+// InsufficientResourceError is an error type that indicates what kind of resource limit is
+// hit and caused the unfitting failure.
+type InsufficientResourceError struct {
```

Comment:
What if we just use `noderesources.InsufficientResource`

Reply:
Given that scheduler don't use `InsufficientResource` to output verbosed log, or rely on the specific shortage amount to make other scheduling decisions, it sounds unnecessary to me.